actors/miner/src/lib.rs

```diff
@@ -3857,91 +3857,83 @@ fn process_early_terminations(
-                        ExitCode::USR_ILLEGAL_STATE,
-                        "failed to pop early terminations",
-                    )
-                })?;
+    let mut sectors_with_data = vec![];
+    let (result, more, penalty, pledge_delta) = rt.transaction(|state: &mut State, rt| {
+        let store = rt.store();
+        let policy = rt.policy();
 
-            // Nothing to do, don't waste any time.
-            // This can happen if we end up processing early terminations
-            // before the cron callback fires.
-            if result.is_empty() {
-                info!("no early terminations (maybe cron callback hasn't happened yet?)");
-                return Ok((result, more, Vec::new(), TokenAmount::zero(), TokenAmount::zero()));
-            }
+        let (result, more) = state
+            .pop_early_terminations(
+                policy,
+                store,
+                policy.addressed_partitions_max,
+                policy.addressed_sectors_max,
+            )
+            .context_code(ExitCode::USR_ILLEGAL_STATE, "failed to pop early terminations")?;
+
+        // Nothing to do, don't waste any time.
+        // This can happen if we end up processing early terminations
+        // before the cron callback fires.
+        if result.is_empty() {
+            info!("no early terminations (maybe cron callback hasn't happened yet?)");
+            return Ok((result, more, TokenAmount::zero(), TokenAmount::zero()));
+        }
 
-            let info = get_miner_info(rt.store(), state)?;
-            let sectors = Sectors::load(store, &state.sectors).map_err(|e| {
-                e.downcast_default(ExitCode::USR_ILLEGAL_STATE, "failed to load sectors array")
-            })?;
+        let info = get_miner_info(rt.store(), state)?;
+        let sectors = Sectors::load(store, &state.sectors).map_err(|e| {
+            e.downcast_default(ExitCode::USR_ILLEGAL_STATE, "failed to load sectors array")
+        })?;
 
-            let mut sectors_terminated: Vec<BitField> = Vec::new();
-            let mut total_initial_pledge = TokenAmount::zero();
-            let mut penalty = TokenAmount::zero();
+        let mut total_initial_pledge = TokenAmount::zero();
+        let mut total_penalty = TokenAmount::zero();
 
-            for (epoch, sector_numbers) in result.iter() {
-                sectors_terminated.push(sector_numbers.clone());
-                let sectors = sectors
-                    .load_sector(sector_numbers)
-                    .map_err(|e| e.wrap("failed to load sector infos"))?;
+        for (epoch, sector_numbers) in result.iter() {
+            let sectors = sectors
+                .load_sector(sector_numbers)
+                .map_err(|e| e.wrap("failed to load sector infos"))?;
 
-                penalty += termination_penalty(
+            for sector in &sectors {
+                total_initial_pledge += &sector.initial_pledge;
+                total_penalty += termination_penalty(
```

Comment:
Since this is the only caller and this is now only two lines consider removing the wrapping function altogether and just calling the power + penalty lines in sequence at the calcite for slight simplicity and efficiency gains.